api/src/main/java/org/openmrs/api/db/hibernate/HibernateAdministrationDAO.java

```diff
@@ -200,9 +204,56 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
 	
 	/**
 	 * @see org.openmrs.api.db.AdministrationDAO#validate(java.lang.Object, Errors)
+	 * @should Pass validation if field lengths are correct
+	 * @should Fail validation if field lengths are not correct
+	 * @should Fail validation for location class if field lengths are not correct
+	 * @should Pass validation for location class if field lengths are correct
 	 */
 	@Override
 	public void validate(Object object, Errors errors) throws DAOException {
+		Class entityClass = object.getClass();
+		String name = object.getClass().getSimpleName();
+		for (Field field : object.getClass().getDeclaredFields()) {
```

Comment:
You don't need to use reflection here, you need to delegate to ValidationUtil when looping over the class metadata below. The ClassMetadata Object has three methods to get information about identifiers, i.e getIndentifierPropertyName(), getIdentifier() and getIndentifierType, so you don't need to use reflection